src/hooks/useVerified.ts

```diff
@@ -34,63 +39,122 @@ const findFailedNft = (response: ChecksResponse) => {
-  roles: string[] = [],
-  checkCallback: any,
-  hasPolygonID: boolean,
-  nftClaimLinks: nftClaimLinksInterface | undefined
-) => {
+const useVerified = ({
+  account,
+  checkCallback,
+  checksIds,
+  roles,
+  nftClaimLinks,
+  hasPolygonID,
+}: IuseVerified) => {
   const [isVerified, setIsVerified] = useState(true);
   const [nftFailed, setNftFailed] = useState('');
   const [status, setStatus] = useState<{
-    someItemFailed: boolean;
-    failedItem: string | undefined;
-    response: KeyBooleanPair;
+    response: KeyBooleanPair | PolygonAuthorizationResponse | any;
+    someItemFailed?: boolean;
+    failedItem?: string | undefined;
+    shouldGetDID?: boolean;
+    showVcModal?: boolean;
   }>({
     someItemFailed: false,
     failedItem: undefined,
+    shouldGetDID: false,
     response: {},
+    showVcModal: false,
   });
   const [apiError, setApiError] = useState('');
   const idsToCheck = checksIds ? checksIds.join(',') : '';
 
-  const checkRoles = async () => {
-    try {
-      const res = await Promise.all(roles.map(async role => doRoleCheck(role)));
+  const callWithDryRun = async (fn: Function, dryRun: boolean) => {
+    const res = await Promise.all(roles.map(async role => fn(role, dryRun)));
 
-      const errorOnCalls = res.find(res => res.error);
+    const errorOnCalls = res.find(res => res.error);
 
-      if (errorOnCalls) {
-        setApiError(errorOnCalls.error);
+    if (errorOnCalls) {
+      if (errorOnCalls?.error === POLYGON_DID_ERROR) {
+        const res = await polyogonAuth();
         setIsVerified(false);
+        setStatus({
+          showVcModal: true,
+          shouldGetDID: true,
+          response: res,
+        });
         return;
       }
 
-      const allRolePassed = res.every(res => res.passed === true);
+      setApiError(errorOnCalls.error);
+      setIsVerified(false);
+      return;
+    }
 
-      if (allRolePassed) {
-        setIsVerified(true);
-        return;
+    return res;
+  };
+
+  const checkRolesDryRunFalse = async () => {
+    try {
+      const res = await callWithDryRun(doRoleCheck, false);
+
+      if (res) {
+        const allRolesSuccess = res.every(res => res.passed === true);
+
+        if (allRolesSuccess) {
+          const vcIdsArray = res.map(res => res.vcs[0]);
+
+          if (!vcIdsArray) {
+            setIsVerified(true);
+            return;
+          }
+
+          setStatus({
+            showVcModal: true,
+            response: { vcIdsArray },
+          });
+          return;
+        }
+
+        setStatus({
+          someItemFailed: true,
+          failedItem: 'Roles',
+          response: {},
+        });
+        setIsVerified(false);
       }
 
-      setStatus({
-        someItemFailed: true,
-        failedItem: 'Roles',
-        response: {},
-      });
+      return;
+    } catch (error) {
+      console.error(`Error on check roles:", ${error}`);
       setIsVerified(false);
+    }
+  };
+
+  const checkRoles = async () => {
```

Comment:
@KyrneDev what do you mean? just create only one big function that takes a boolean for dryrun?
but if i need to do an other call ? I need a function to just make the call to the endpoint so that inside the checkRoles function I can call it as many times as I want.

Reply:
The only difference between those two functions is these lines:

```
if (!vcIdsArray) {
            setIsVerified(true);
            return;
}
          ```

Combine if you can, if not, no worries

Reply:
let me see thanks

Reply:
done!